fetch

```diff
@@ -2,24 +2,44 @@
 
 # API documentation: http://www.mediawiki.org/wiki/API:Main_page
 
-. wiki-config
- 
+source wiki-config
+
 TITLE="$1"
 
-if [ -z "$TITLE" -o -n "$2" ]; then
-  echo "Syntax: $0 <page_title>"
+if [ -z "${TITLE:-}" -o -n "${2:-}" ]; then
+  echo "ERROR: Syntax: $0 <page_title>"
   exit 1
 fi
 
-exec curl -s -S \
+## NOTE: Do not use 'echo' here as this script writes to stdout which is
+##       processed by other scripts.
+true "INFO: Fetching '${WIKIINDEX}' '$TITLE'..."
+
+# curl \
+# 	--fail \
+#    --silent \
+#    --show-error \
+# 	--location \
+# 	--cookie "$cookie_jar" \
+# 	--cookie-jar "$cookie_jar" \
+# 	--user-agent "Curl Shell Script" \
+# 	--keepalive-time 60 \
+# 	--header "Accept-Language: en-GB" \
+# 	--header "Connection: keep-alive" \
+# 	--header "Expect:" \
+# 	--data-urlencode "title=$TITLE" \
+# 	--request "POST" "${WIKIINDEX}?action=raw"
+
+curl \
+   --fail \
+   --silent \
+   --show-error \
 	--location \
-	--cookie $cookie_jar \
-	--cookie-jar $cookie_jar \
```

Comment:
My site specific use case. Avoiding accidental download of non-public pages.
(Which are non-public, require login? For example history. Otherwise bots are hammering the server, following all links possible, comparing every  history revision with every revision.)